File: fragment/templates/app-java/src/main/java/_Presenter.java
package <%= appPackage %>.ui.<%= fragmentPackageName %>;

<% if (componentType == 'createNew') { %>import <%= appPackage %>.di.ActivityScope;<% } %>
import <%= appPackage %>.ui.base.BasePresenter;

import javax.inject.Inject;
<% if (componentType == 'useApplication') { %>import javax.inject.Singleton;<% } %>

<% if (componentType == 'createNew') { %>@ActivityScope<% } else if (componentType == 'useApplication') { %>@Singleton<% } %>
public class <%= fragmentName %>Presenter extends BasePresenter<<%= fragmentName %>View> {

    @Inject
    public <%= fragmentName %>Presenter() {

    }

}

File: fragment/templates/app-kotlin/src/main/java/_Presenter.kt
package <%= appPackage %>.ui.<%= fragmentPackageName %>;

<% if (componentType == 'createNew') { %>import <%= appPackage %>.di.ActivityScope;<% } %>
<% if (componentType == 'useApplication') { %>import javax.inject.Singleton;<% } %>
import <%= appPackage %>.ui.base.BasePresenter;

import javax.inject.Inject;

<% if (componentType == 'createNew') { %>@ActivityScope<% } else if (componentType == 'useApplication') { %>@Singleton<% } %>
class <%= fragmentName %>Presenter @Inject constructor() : BasePresenter<<%= fragmentName %>View>() {

}
